File: docs/scripts_js/ArcInvestigation.js
import { ArcInvestigation, Comment$ as Comment, XlsxController, JsonController} from "@nfdi4plants/arctrl"
// Import Spreadsheet to XLSX reader/writer
import {Xlsx} from "@fslab/fsspreadsheet";

// # Comments

const investigation_comments = ArcInvestigation.init("My Investigation")

const newComment = new Comment("The Name", "The Value")
const newComment2 = new Comment("My other Name", "My other Value")

investigation_comments.Comments.push(newComment)
investigation_comments.Comments.push(newComment2)

console.log(investigation_comments)

// # IO

// ## XLSX - Write

let fswb = XlsxController.Investigation.toFsWorkbook(investigation_comments)

console.log(fswb)

// Xlsx.toFile("test.isa.investigation.xlsx", fswb)

// Json - Write

const investigation = ArcInvestigation.init("My Investigation")

const json = JsonController.Investigation.toJsonString(investigation)

console.log(json)

// Json - Read

const jsonString = json

const investigation_2 = JsonController.Investigation.fromJsonString(jsonString)

console.log(investigation_2.Equals(investigation))

File: docs/scripts_python/ArcInvestigation.py
from arctrl.arctrl import ArcInvestigation, Comment, XlsxController, JsonController
from fsspreadsheet.xlsx import Xlsx

# Comments
investigation_comments = ArcInvestigation.init("My Investigation")

new_comment = Comment("The Name", "The Value")
new_comment2 = Comment("My other Name", "My other Value")

investigation_comments.Comments.append(new_comment)
investigation_comments.Comments.append(new_comment2)

print(investigation_comments)

# IO

# XLSX - Write
fswb = XlsxController.Investigation().to_fs_workbook(investigation_comments)

Xlsx.to_xlsx_file("test.isa.investigation.xlsx", fswb)

# Json - Write
investigation = ArcInvestigation.init("My Investigation")

json_str = JsonController.Investigation().to_json_string(investigation)

print(json_str)

# Json - Read
json_string = json_str

investigation_2 = JsonController.Investigation().from_json_string(json_string)

print(investigation_2.__eq__(investigation))